lib/cryptoexchange/exchanges/okex_swap/services/contract_stat.rb

```diff
@@ -0,0 +1,41 @@
+module Cryptoexchange::Exchanges
+  module OkexSwap
+    module Services
+      class ContractStat < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            true
+          end
+        end
+
+        def fetch(market_pair)
+          unless market_pair.contract_interval == "perpetual"
+            open_interest = super(open_interest_url(market_pair))
+            index = super(index_url(market_pair))
+            adapt(open_interest, index, market_pair)
+          end
+        end
+
+        def open_interest_url(market_pair)
+          "https://www.okex.com/api/futures/v3/instruments/#{market_pair.inst_id}/open_interest"
```

Comment:
If the URL domain/prefix are shared from the `market.rb`,you can use that constant

Reply:
they are not the same.. anyway, i have assign them to a `CONTRACT_STAT_URL` as it is repetitive